openedx/features/course_card/views.py

```diff
@@ -66,8 +69,17 @@ def get_course_cards(request):
     )
 
 
-def get_course_with_link_and_start_date(course, course_rerun_object, request):
+def get_course_dict(course, programs):
+    course_dict = vars(course)
+
+    course_dict['display_name_with_default'] = course.display_name_with_default
+    course_dict['display_number_with_default'] = course.display_number_with_default
+    course_dict['specialization'] = check_course_part_of_specialization(course.id, programs)
```

Comment:
Can we please change the `specialization` to `is_program_course`

Reply:
I have a approach in my mind to avoid passing programs, As concerned programs will always be all programs. So instead of passing we can get programs inside the method `check_course_part_of_specialization` by using 
```
from crum import get_current_request
request = get_current_request()
programs = get_programs(request.site)
```

It ll help us to simplify the code by avoiding passing programs in get_course_dict and then check_course_part_of_specialization and decrease the number of parameters.

It would be great if we can use this.

Reply:
@zeshanarifios actually that function is getting called inside a loop. if we take `get_programs` inside `check_course_part_of_specialization` then it will retrieve data from cache against each course. according to current approach it retrieve it only once.

Reply:
@monofal 👍 
How about calling it inside `get_course_dict`, It ll decrease 1 level and if it needs to be called anywhere else we don't need to pass it again.